packages/api/src/middleware/validation/checkAnimalOrBatch.js

```diff
@@ -171,78 +171,71 @@ const checkAnimalBreed = async (
-    setFalsyValuesToNull(breedKeyOptions, animalOrBatch);
-  }
-
-  if (
-    someExists(breedKeyOptions, animalOrBatch) &&
-    !someTruthy([default_breed_id, custom_breed_id, breed_name])
-  ) {
-    // do nothing if nulling breed
-  } else {
-    // Check if default breed or default type is present
-    if (
-      (someExists(breedKeyOptions, animalOrBatch) && default_breed_id) ||
-      (someExists(typeKeyOptions, animalOrBatch) && default_type_id)
-    ) {
-      await checkDefaultBreedMatchesType(
-        preexistingAnimalOrBatch,
-        default_breed_id,
-        default_type_id,
+  // If neither breed or type is specified, skip checks
+  if (someExists(breedKeyOptions, animalOrBatch) || someExists(typeKeyOptions, animalOrBatch)) {
+    // Check only one breed option is truthy
+    if (someTruthy([default_breed_id, custom_breed_id, breed_name])) {
+      checkExactlyOneIsProvided(
+        [default_breed_id, custom_breed_id, breed_name],
+        'default_breed_id, custom_breed_id, or breed_name',
       );
     }
-    // Check if custom breed or custom type is present
-    if (
-      (someExists(breedKeyOptions, animalOrBatch) && custom_breed_id && !type_name) ||
-      (someExists(typeKeyOptions, animalOrBatch) &&
-        (default_type_id || custom_type_id) &&
-        !breed_name)
-    ) {
-      let customBreed;
-      // Find customBreed if exists
-      if (someExists(breedKeyOptions, animalOrBatch) && custom_breed_id) {
-        checkIdIsNumber(custom_breed_id);
-        customBreed = await CustomAnimalBreedModel.query()
-          .whereNotDeleted()
-          .findById(custom_breed_id);
-        if (!customBreed) {
-          throw customError('Custom breed does not exist');
-        }
-      } else if (preexistingAnimalOrBatch?.custom_breed_id) {
-        checkIdIsNumber(preexistingAnimalOrBatch?.custom_breed_id);
-        customBreed = await CustomAnimalBreedModel.query()
-          .whereNotDeleted()
-          .findById(preexistingAnimalOrBatch.custom_breed_id);
-        if (!customBreed) {
-          // This should not be possible
-          throw customError('Custom breed does not exist');
-        }
-      }
-      // Check custom breed if exists
-      if (customBreed) {
-        await checkRecordBelongsToFarm(customBreed, farm_id, 'custom breed');
-        checkCustomBreedMatchesType(
-          animalOrBatch,
+    // Overwrite all others with null in db if editing, post does not accept nulled values in oneOf schemas
+    if (!creating && someExists(breedKeyOptions, animalOrBatch)) {
+      setFalsyValuesToNull(breedKeyOptions, animalOrBatch);
+    }
+
+    const isNotNullingAllBreedOptions = !(
+      someExists(breedKeyOptions, animalOrBatch) &&
+      !someTruthy([default_breed_id, custom_breed_id, breed_name])
+    );
+    const isCreatingWithBreed =
+      creating && someTruthy([default_breed_id, custom_breed_id, breed_name]);
+    // Do checks on breed unless removing breed from existing record or
```

Comment:
"Do checks on breed unless removing breed from existing record" - it seems to me that the next line can simply be `if(someTruthy([default_breed_id, custom_breed_id, breed_name])`.
I got confused here and I may have misunderstood something.

Reply:
So if you comment out the logic and replace it with the suggested you will see that 3 tests fail. The tests that fail are all tests that edit the `type`. They fail because they successfully allow the user to change the type to one that does not match the breed (eg honey bee breed and cow type would be an allowed combination). 

So I will fix/update the comment since I forgot to finish the sentence. 
> Do checks on breed unless removing breed from existing record or creating animal without breed

This describes when it should not run the checks but can also be read as: 

> Check breed when creating new animal with breed or when editing an animal that has type or breed being edited.. but not when the breed is being nulled because you don't need to check it if you are nulling it.

It was a logic structure decision to make breed tests be more complex. Since breeds must have types but types do not have to have breeds. It is equally valid ways to structure the logic so that breed tests were mixed into the type middleware, but I found that hard to understand.

Please let me know if you have suggestions for better comments!

Reply:
Ah, I forgot about the existing breed! The previous comment should've been clear enough. Apologies for the misunderstanding!